reflex/state.py

```
@@ -1233,6 +1236,18 @@ def __setattr__(self, name: str, value: Any):
             self._mark_dirty()
             return
 
+        if (
+            not name.startswith(
+                "_"
+            )  # TODO: skipping backend vars and vars with double leading underscores for now. They should be supported, however.
+            and name not in self.vars
+            and name not in self.get_skip_vars()
+        ):
+            raise AttributeError(
```

I think we should raise a SetUndefinedStateVar error instead.